silverback/_cli.py

```diff
@@ -230,30 +322,201 @@ def new_cluster(
+        raise click.UsageError("Cluster is not funded via ApePay Stream")
+
+    elif token_amount is None and stream_time is None:
+        raise click.UsageError("Must specify one of '--amount' or '--time'.")
+
+    if not token_amount:
+        assert stream_time  # mypy happy
+        one_token = 10 ** stream.token.decimals()
+        token_amount = int(
+            one_token
+            * (
+                stream_time.total_seconds()
+                / stream.manager.compute_stream_life(
+                    account.address, stream.token, one_token, stream.products
+                ).total_seconds()
+            )
+        )
+
+    if not stream.token.balanceOf(account) >= token_amount:
+        raise click.UsageError("Do not have sufficient funding")
+
+    elif not stream.token.allowance(account, stream.manager.address) >= token_amount:
+        click.echo(f"Approving StreamManager({stream.manager.address})")
+        stream.token.approve(
+            stream.manager.address,
+            2**256 - 1 if click.confirm("Unlimited Approval?") else token_amount,
+            sender=account,
+        )
+
+    click.echo(
+        f"Funding Stream for Cluster '{cluster_path}' with "
+        f"{token_amount / 10**stream.token.decimals():0.4f} {stream.token.symbol()}"
+    )
+    stream.add_funds(token_amount, sender=account)
+
+    click.echo(f"{click.style('SUCCESS', fg='green')}: Cluster funded for {stream.time_left}.")
+
+
+@pay.command(name="cancel", cls=ConnectedProviderCommand)
+@account_option()
+@click.argument("cluster_path", metavar="CLUSTER")
+@platform_client
+def cancel_payment_stream(
+    platform: PlatformClient,
+    network: NetworkAPI,
+    account: AccountAPI,
+    cluster_path: str,
+):
+    """
+    Shutdown CLUSTER and refund all funds to Stream owner
+
+    NOTE: Only the Stream owner can perform this action!
+    """
+
+    if "/" not in cluster_path or len(cluster_path.split("/")) > 2:
+        raise click.BadArgumentUsage(f"Invalid cluster path: '{cluster_path}'")
+
+    workspace_name, cluster_name = cluster_path.split("/")
+    if not (workspace_client := platform.workspaces.get(workspace_name)):
+        raise click.BadArgumentUsage(f"Unknown workspace: '{workspace_name}'")
+
+    elif not (cluster := workspace_client.clusters.get(cluster_name)):
+        raise click.BadArgumentUsage(
+            f"Unknown cluster in workspace '{workspace_name}': '{cluster_name}'"
+        )
+
+    elif cluster.status != ResourceStatus.RUNNING:
+        raise click.UsageError(f"Cannot fund '{cluster_info.name}': cluster is not running.")
+
+    elif not (stream := workspace_client.get_payment_stream(cluster, network.chain_id)):
+        raise click.UsageError("Cluster is not funded via ApePay Stream")
+
+    if click.confirm(
+        click.style("This action is irreversible, are you sure?", bold=True, bg="red")
+    ):
+        stream.cancel(sender=account)
```

Comment:
I just want to confirm that we're properly calculating the payback here. This is after fees and usage, correct? I'm trying to dig through the code here, but it's layered too deep to do it github to find it simply. I'm leaving a note to be sure we're properly managing this cancel/refund

Reply:
The funds stream to us at a defined rate when the stream is created. When cancel is called on the stream, the remaining funds are refunded back to the owner